python/lsst/analysis/drp/plotUtils.py

```diff
@@ -105,3 +107,163 @@ def addPlotInfo(fig, plotInfo):
+
+    Parameters
+    ----------
+    xs : `numpy.ndarray`
+        The color on the xaxis
+    ys : `numpy.ndarray`
+        The color on the yaxis
+    paramDict : lsst.pex.config.dictField.Dict
+        A dictionary of parameters for line fitting
+        xMin : `float`
+            The minimum x edge of the box to use for initial fitting
+        xMax : `float`
+            The maximum x edge of the box to use for initial fitting
+        yMin : `float`
+            The minimum y edge of the box to use for initial fitting
+        yMax : `float`
+            The maximum y edge of the box to use for initial fitting
+        mHW : `float`
+            The hardwired gradient for the fit
+        bHW : `float`
+            The hardwired intercept of the fit
+
+    Returns
+    -------
+    paramsOut : `dict`
+        A dictionary of the calculated fit parameters
+        xMin : `float`
+            The minimum x edge of the box to use for initial fitting
+        xMax : `float`
+            The maximum x edge of the box to use for initial fitting
+        yMin : `float`
+            The minimum y edge of the box to use for initial fitting
+        yMax : `float`
+            The maximum y edge of the box to use for initial fitting
+        mHW : `float`
+            The hardwired gradient for the fit
+        bHW : `float`
+            The hardwired intercept of the fit
+        mODR : `float`
+            The gradient calculated by the ODR fit
+        bODR : `float`
+            The intercept calculated by the ODR fit
+        yBoxMin : `float`
+            The y value of the fitted line at xMin
+        yBoxMax : `float`
+            The y value of the fitted line at xMax
+        bPerpMin : `float`
+            The intercept of the perpendicular line that goes through xMin
+        bPerpMax : `float`
+            The intercept of the perpendicular line that goes through xMax
+        mODR2 : `float`
+            The gradient from the second round of fitting
+        bODR2 : `float`
+            The intercept from the second round of fitting
+        mPerp : `float`
+            The gradient of the line perpendicular to the line from the
+            second fit
+
+    Notes
+    -----
+    The code does two rounds of fitting, the first is initiated using the
+    hardwired values given in the `paramDict` parameter and is done using
+    an ODR fit to the points defined by the box of xMin, xMax, yMin and
+    yMax. Once this fitting has been done a perpendicular bisector is
+    calculated at either end of the line and only points that fall within
+    these lines are used to recalculate the fit.
+    """
+
+    # Points to use for the fit
+    fitPoints = np.where((xs > paramDict["xMin"]) & (xs < paramDict["xMax"])
+                         & (ys > paramDict["yMin"]) & (ys < paramDict["yMax"]))[0]
+
+    def linearFit(B, x):
+        return B[0]*x + B[1]
+
+    linear = scipyODR.Model(linearFit)
+    linear = scipyODR.polynomial(1)
```

Comment:
Don't you want to choose one or the other? This appears to be defining "linear" twice.